src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -200,9 +204,9 @@ private void PopulateAssemblies()
         /// This is not meant to be a general mechanism for injecting files that can't be handled
         /// through the normal NuGet package mechanisms.
         /// </summary>
-        private void PopulatePreResolvedNetStandardAssemblies()
+        private void PopulateExistingReferenceItems()
         {
-            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            var existingReferenceItemDependencies = new List<ITaskItem>();
             foreach (var reference in References)
             {
                 var packageName = reference.GetMetadata("NuGetPackageId");
```

Comment:
Yeah, this loop is going to get all the references that came from NuGet as well.  Right now those don't get excluded until below: 
>// If we already know about the assembly file, skip it.

I was thinking you could have a fast path that avoids some of this work.  I looked at how you determined *already know about* and saw that this was based on [Type metadata](https://github.com/dotnet/sdk/pull/1352/files#diff-56c31b357aa781955f3963adda6c22d2R184), I noticed that metadata is preserved on the `Reference` items, so I thought you could fast-path the check rather than doing the lookups and string comparison below.